File: EDA/src/eda/Hanoi.java
import java.util.Scanner;
class Hanoi {
	public void move(int n, String origem, String aux, String destino){
		if (n >= 1){
			move(n - 1, origem, destino, aux);
			//desempilha da origem e empilha no destino
			System.out.printf("Move o disco %d da haste %s para haste %s\n", 
								n, origem, destino); 
			move(n - 1, aux, origem, destino);
		}
	}

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		Hanoi hanoi = new Hanoi();
		int n = sc.nextInt();
		hanoi.move(n, "A", "B", "C");
	}
}

File: EDA/src/eda/hanoi.py
def hanoi(n, A, B, C):
    if (n >= 1):
        hanoi( n - 1, A, C, B)
        print("Move o disco %d da haste %s para a haste %s" %(n, A, C))
        hanoi( n - 1, B, A, C)

altura = int(raw_input())
hanoi(altura, "A", "B", "C")
